datacube_stats/utils/__init__.py

```diff
@@ -104,6 +116,22 @@ def da_nodata(da, default=None):
     return 0
 
 
+def nodata_like(ds):
+    """Similar to xarray.full_like but filled with nodata value or with NaN for
+    floating point variables.
+
+    """
+    assert isinstance(ds, (xarray.Dataset, xarray.DataArray))
+
+    def _nodata_like_da(da):
+        return xarray.full_like(da, da_nodata(da))
+
+    if isinstance(ds, xarray.DataArray):
+        return _nodata_like_da(ds)
+
+    return ds.apply(_nodata_like_da, keep_attrs=True)
+
+
```

Comment:
Will probably work, but it's not obvious how recursion will stop. I know that it will, because of the way `ds.apply` works, but it's not obvious by just looking at it.

Reply:
Recursion will stop at a `DataArray`. `Dataset`s are just collection of `DataArray`s.